uaclient/api/u/pro/security/vulnerabilities/_common/v1.py

```diff
@@ -0,0 +1,325 @@
+
+    def __init__(
+        self,
+    ):
+        self.vulnerabilities = {}
+
+    def get_package_vulnerabilities(self, affected_pkg: Dict[str, Any]):
+        raise NotImplementedError
+
+    def parse_data(
+        self,
+        vulnerabilities_data: Dict[str, Any],
+        installed_pkgs_by_source: Dict[str, Dict[str, str]],
+    ):
+        affected_pkgs = vulnerabilities_data.get("packages", {})
+        vulns_info = vulnerabilities_data.get("security_issues", {}).get(
+            self.vulnerability_type, {}
+        )
+
+        for source_pkg, binary_pkgs in installed_pkgs_by_source.items():
+            affected_pkg = affected_pkgs.get(source_pkg, {})
+            source_version = affected_pkg.get("source_versions", {})
+
+            for vuln_name, vuln in self.get_package_vulnerabilities(
+                affected_pkg
+            ).items():
+                vuln_info = vulns_info.get(vuln_name, "")
+                vuln_fixed_version = vuln.get("source_fixed_version")
+                vuln_status = vuln.get("status")
+
+                # if the vulnerability fixed version is None,
+                # that means that no fix has been published
+                # yet.
+                if vuln_fixed_version is None:
+                    if vuln_status != "not-vulnerable":
+                        if vuln_name not in self.vulnerabilities:
+                            self.vulnerabilities[vuln_name] = vuln_info
+                            self.vulnerabilities[vuln_name][
+                                "affected_packages"
+                            ] = []
+
+                        self.vulnerabilities[vuln_name][
+                            "affected_packages"
+                        ].extend(
+                            {
+                                "name": pkg_name,
+                                "current_version": pkg_version,
+                                "fix_version": None,
+                                "status": vuln_status,
+                                "fix_available_from": None,
+                            }
+                            for pkg_name, pkg_version in sorted(
+                                binary_pkgs.items()
+                            )
+                        )
+
+                    continue
+
+                for pkg_name, pkg_version in sorted(binary_pkgs.items()):
+                    try:
+                        pocket = source_version[vuln_fixed_version].get(
+                            "pocket"
+                        )
+                        fix_version = (
+                            source_version[vuln_fixed_version]
+                            .get("binary_packages", {})
+                            .get(pkg_name, "")
+                        )
+                    except KeyError:
+                        # There is bug in the data where some sources are
+                        # not present. The Security team is already aware
```

Comment:
Do we still need this? If we do is there a reference to a bug or something that will be updated when it is fixed?

Reply:
It seems that we do, but even though I will talk with the Security team about this, I still think we should keep this just to be safe in the future